File: algorithms/0053.Maximum Subarray/Solution.java
public class Solution {

  public int maxSubArray(int[] nums) {
    int sum = 0;
    int max = Integer.MIN_VALUE;
    for (int i = 0; i < nums.length; i++) {
      int n = nums[i];
      sum += n;
      max = Math.max(max, sum);
      if (sum < 0) {
        sum = 0;
      }
    }
    return max;
  }
}

File: algorithms/0053.Maximum Subarray/solution.go
package main

func maxSubArray(nums []int) int {
	cur, res := nums[0], nums[0]
	for _, n := range nums[1:] {
		cur = max(n, cur+n)
		res = max(res, cur)
	}
	return res
}

func max(n1, n2 int) int {
	if n2 > n1 {
		return n2
	}
	return n1
}
